src/utilities/compareVersions.js

```diff
@@ -0,0 +1,69 @@
+import { Client as BugsnagClient } from 'bugsnag-react-native';
+const bugsnagClient = new BugsnagClient();
+
+export function compareVersions(versionOne, versionTwo) {
+  try {
+    const result = versionToInteger(versionOne) - versionToInteger(versionTwo);
+    if (result > 0) return 1;
+    if (result < 0) return -1;
+    return 0;
+  } catch (error) {
+      // Errors thrown in migration method in dataMigration.js do not trickle down to bugsnag
+      // so will need manually notify bug snag.
+    error.message = `error verifying versions, versionOne: ${versionOne}, versionTwo ${versionTwo}`;
+    bugsnagClient.notify(error);
+  }
+  return 0;
+}
+// versionToInteger logic is the same as ./android/app/build.gradle
+function versionToInteger(version) {
+  const [majorMinorPatch, provisional] = version.split('-');
+  const [major, minor, patch] = majorMinorPatch.split('.');
+  return Number(major) * 10000000 +
+         Number(minor) * 100000 +
+         Number(patch) * 100 +
+         Number(!provisional ? '99' : provisional.match(/[0-9]+/)[0]);
+         // Regex /[0-9]+/ matches only numbers and at least on digit number
+}
+
+export function test() {
+  const testHelper = (versionOne, versionTwo, expectedResult) => {
+    const testResult = compareVersions(versionOne, versionTwo);
+    const testMessage = testResult === expectedResult ? 'pass' : 'FAIL';
+    console.log(`checking ${versionOne} vs ${versionTwo}, status: ${testMessage}`);
+    console.log(`expected result: ${expectedResult}, actual result: ${testResult}`);
+  };
+
+  testHelper('1.0.0', '1.0.1', -1);
+  testHelper('1.0.0', '1.1.0', -1);
+  testHelper('1.0.0', '2.0.0', -1);
+  testHelper('2.0.0', '2.0.0', 0);
```

Comment:
Missed some example of
testHelper('2.0.0', '1.0.0', 1);

Reply:
Why assume when you can test 😛

Reply:
Sure testing is the key. To me it seemed like 1 + 2 = 2 + 1